packages/host/app/components/matrix/room.gts

```diff
@@ -250,6 +251,9 @@ export default class Room extends Component<Signature> {
         messageElemements: new WeakMap(),
         messageScrollers: new Map(),
         messageVisibilityObserver: new IntersectionObserver((entries) => {
+          if (isDestroyed(this)) {
```

Comment:
I was getting the `You attempted to get the value of a helper after the helper was destroyed, which is not allowed"` 
 error in the `Commands tests: a command sent via SendAiAssistantMessageCommand without autoExecute flag is not automatically executed by the bot` test.
 
My guess is that when the AI panel is opened in interactive mode and a switch to code mode happens, the room gets rerendered (due to operator mode state change) and the intersection observer wants to work on the previous room rendition but it can't since it was destroyed.

Reply:
Found a better approach: https://github.com/cardstack/boxel/pull/2076/commits/47c35a11cf1b2736b9608b0684567e13def249d4